### ISSUE TITLE
Focus Appearance (Minimum), minumum area, second calculation could be too small

### ISSUE THREAD

**Issue by mbgower** [w3c/wcag#1684]
The first part of the current editor's draft of 2.4.11 reads as follows:

When user interface components receive keyboard focus, all of the following are true:
- Minimum area: The focus indicator area is either:
  - at least as large as the area of a 1 CSS pixel thick perimeter of the unfocused component;
  - at least as large as a 4 CSS pixels border along the shortest side of the unfocused component, and no thinner than 2 CSS pixels.

Although I believe the intent of the second sub-bullet is to say that the focus indicator area along the shortest side is as large as a 4 CSS pixel border along that side, it can be misinterpreted to mean that the indicator is as large as 4 x the shortest side, but the indicator can be applied anywhere.

[Figure 3 of the current editor's draft](https://w3c.github.io/wcag/understanding/focus-appearance-minimum.html#focus-indicator-underline) shows an example of meeting the first sub-bullet
![image](https://user-images.githubusercontent.com/14298245/111323564-3bcbf180-8627-11eb-9444-b0986a11d79b.png)

However, it can also be used to illustrate how the second sub-bullet could be interpreted to accept a smaller focus indicator.

In this interpretation, the shortest side of the control is 30 CSS pixels. Four times this value is 120 CSS pixels. Using this value, I can now create an indicator of only 120 CSS pixels and meet the Minimum area bullet. (There is nothing in the second sub bullet stating the indicator must be on the short side only). I could place a 60 CSS pixel long line anywhere on the control, as long as it was 2 CSS pixels thick, and it appears to meet the requirement. Here's a very rough edit of the existing figure 1 to show how it appears to meet sub-bullet 2. Meets like this: 30 CSS Pixel x 2 = 60 CSS pixels on the side, and the same 30 x 2 on a third of the bottom side, for a total of 120 CSS pixels.
![image](https://user-images.githubusercontent.com/14298245/111325839-4f785780-8629-11eb-90c5-848586465e0e.png)

I think this could be resolved by rewording the sub-bullet, although I'm not entirely sure I can parse exactly how the sub-bullet is currently intended.

**Comment by mraccess77** [comment]
Hi @mbgower Yes, that is my understanding, and from what I understood intentional. I have raised that this change also has no contrast requirement with adjacent pixels - but does with changed pixels. https://github.com/w3c/wcag/issues/1646

**Comment by JAWS-test** [comment]
I think there should be no rules about where the focus indicator is located and what it looks like, the main thing is that it is large enough and has sufficient contrast.

Also, the focus indicator should be assignable to the element. For this requirement there were some issues, but there was no solution.  That is why this has not been solved yet.

**Comment by alastc** [comment]
> In this interpretation, the shortest side of the control is 30 CSS pixels. Four times this value is 120 CSS pixels. Using this value, I can now create an indicator of only 120 CSS pixels and meet the first bullet.

That is correct, if you look at the comments on the issue Jon raised (#1646), and examples from: 
- https://codepen.io/alastc/pen/JjbNEKz
- https://alastairc.uk/tests/wcag22-examples/focus-more-visible-5.html

I don't see another way of allowing for the short & thick examples without that option.

**Comment by mbgower** [comment]
How about making it a multiple of the short side? This could be expressed in bullet form:

- Four times the area of a 1 CSS pixel thick border along the shortest side of the unfocused component, allowable only when the focus indicator area is restricted to an enhancement of the shortest side.

Or we could try to work it in as an exception, and merge the first sub-bullet into the first bullet:
> - **Minimum area**: The focus indicator area is as large as the area of a 1 CSS pixel thick perimeter of the unfocused component.

>Exception: Where the indicator is restricted to the short side of the component, the focus indicator area may be calculated as 4 times the length of the short side, measured in CSS pixels.

**Comment by alastc** [comment]
Perhaps, I need to consider the various examples we've been through.

Another option would be to say that the min thickness is 4px:
"at least as large as a 4 CSS pixels border along the shortest side of the unfocused component, and no thinner than 4 CSS pixels."

**Comment by mbgower** [comment]
Oh, I see. So your "no thinner" is an attempt to force it to be a big blob, no matter where it is placed? I didn't understand that -- and I definitely didn't understand it in the current form where it is only 2 CSS pixels. That 2 CSS pixels seems to create a lot of loopholes!
As @JAWS-test states:
> the main thing is that it is large enough...

Do you think my rendering of something that meets the second sub-bullet (a 2 CSS pixel line that covers one short side and a third of the bottom side) is sufficient? Since it's half of what the first bullet requires, it seems pretty poor. Being super lazy here and just flipping it upside down so that the same treatment is now on the right side and upper right corner. Depending on the magnification level (and the size of the button) is it possible a low vision user wouldn't even have any part of the indicator in their viewport? It certainly wouldn't be as distinguishable as a 2 px border around the whole thing.
![Screen Shot 2021-03-16 at 7 19 35 AM](https://user-images.githubusercontent.com/14298245/111478409-4b157280-86ed-11eb-9acd-e3d14943d1a2.png)

**Comment by JAWS-test** [comment]
@mbgower 

If I understand 2.4.11 correctly, there are no specifications to the focus indicator in terms of:
* Position
* Shape
* Spreading

I.e. the focus indicator can be anywhere, can be a line, a circle, an icon, an arbitrarily shaped area, can consist of letters (e.g. only the text color of the button changes), i.e. the focus indicator does not have to form a contiguous area either.

The only important thing is: the sum of all pixels forming the focus indicator must meet the contrast and size requirements.

Your example therefore fulfills the SC.

By the way, I don't think it is a big problem if the focus indicator is not visible at a large zoom, because the focused element is automatically scrolled into the visible area, i.e. I know where the focus is even without the focus indicator

**Comment by alastc** [comment]
> To me, it has basically cut the Minimum area in half.

Well, that depends on the shape, the maths may not be not intuitive here! E.g. for a square it is exactly the same requirement (+4px): https://alastairc.uk/tests/wcag22-examples/focus-more-visible-5.html

If you have a look through those examples, you might see why I'm leaning more towards the 4px min thickness, otherwise example 7 really couldn't pass.

@JAWS-test wrote:
> If my understanding is wrong...

Nope, I think you got it.

**Comment by alastc** [comment]
Hi Folks, given that we agreed with the current (slightly updated) text, I assume it is ok to close this one?
Please re-open if not...
